### PR TITLE
Add YOLOv8 ONNX C++ inference Example

### PR DESCRIPTION
**Pull request by JustasBart**
Dropped in the yolov8/yolov5 CPP Inference example.

This is my first PR so this might take a bit of patience...

I'm willing to make any changes as required to proceed with this PR :rocket: 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Integration of YOLOv8/YOLOv5 ONNX Inference Example in C++

### 📊 Key Changes
- Added a new C++/ONNX example for performing inference with YOLOv8 or YOLOv5 models.
- Created a `CMakeLists.txt` for building the example using CMake.
- Supplied a detailed `README.md` on how to set up, build, and run the example, and how to export YOLOv8/YOLOv5 models.
- Provided C++ source files (`inference.cpp`, `inference.h`, `main.cpp`) to handle model loading, inference, and post-processing.
- Removed a dummy inventory management example.

### 🎯 Purpose & Impact
- **Purpose**: To offer an example of how to perform inference using YOLOv8/YOLOv5 models in a C++ environment, potentially with GPU acceleration using CUDA.
- **Impact**: Developers can now reference this example to understand how to deploy YOLO models within their C++ applications, which could enhance performance in production environments. For users, this means faster and more efficient object detection capabilities in software that utilizes these models. 🚀

### CODE REVIEW

**Comment by AyushExel** [comment]
@JustasBart you you comment "I have read the CLA Document and I sign the CLA" on this thread ?

**Comment by AyushExel** [comment]
@JustasBart a few things that need updating:
* We can get rid of the data/ folder. Users can just use the ultralytics/assets folder
* We can also get rid of the models folder. Just add the command to export the model in the README itself

**Comment by JustasBart** [comment]
@AyushExel Wasn't aware of the CLA at all, it's all news to me, will resolve this today and will make the changes as suggested, thanks!

**Comment by JustasBart** [comment]
I have read the CLA Document and I sign the CLA.

**Comment by JustasBart** [comment]
@AyushExel I'm still not too sure how to handle the classes.txt file, should I perhaps hard-code the class names as a vector within the main.cpp?

**Comment by AyushExel** [comment]
> @AyushExel I'm still not too sure how to handle the classes.txt file, should I perhaps hard-code the class names as a vector within the main.cpp?

Yeah that sounds good

**Comment by JustasBart** [comment]
@AyushExel It's ready to be reviewed again, I don't think that there is much else to be cut-away anymore... Let me know :rocket:

**Comment by AyushExel** [comment]
Okay this looks good :) 
I'll add it to examples on friday

File: examples/Yolov8_CPP_Inference/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# yolov8/yolov5 Inference C++
+
+Usage:
+
+```
+git clone https://github.com/JustasBart/yolov8_CPP_Inference_OpenCV_ONNX
+cd yolov8_CPP_Inference_OpenCV_ONNX
+```
+
```
Comment (AyushExel) [inline examples/Yolov8_CPP_Inference/README.md:9]: @JustasBart we need to use ultralytics repo for this. Same for other commands. 
```
# git clone ultralytics
pip install .
cd examples/cpp_
...
```

**Comment by JustasBart** [comment]
@AyushExel Changed the paths, thanks!

**Comment by AyushExel** [comment]
Thanks. One final thing, can you also add your example with the proper title in this table here https://github.com/ultralytics/ultralytics/tree/main/examples

**Comment by JustasBart** [comment]
@AyushExel I'm not sure how it's meant to be indented, but I've updated the actual information, thanks!

**Comment by AyushExel** [comment]
@JustasBart perfect looks good. Now we can redirect users with ONNX export-related questions here.
@glenn-jocher I'm merging this but these examples need more visibility otherwise users will keep asking the same questions. I've done something similar in this PR to highlight this a bit - https://github.com/ultralytics/ultralytics/pull/845

**Pull request merged**